File: src/controle-fluxo-empresarial/src/models/Compras/CompraProduto.ts
import { IBaseEntity2 } from './../BaseEntity';
import { Produto } from './../Movimentos/Produto';

export interface CompraProduto extends IBaseEntity2{

    produto?: Produto | null;
    produtoId?: number | null;

    quantidade?: number | null;
    valorUnitario?: number | null;
    desconto?: number | null;
    ipi?: number | null;
    custoUnitario?: number | null;
}

File: src/ControleFluxoEmpresarial/Models/Compras/CompraProduto.cs
using ControleFluxoEmpresarial.Entities;
using ControleFluxoEmpresarial.Models.Movimentos;

namespace ControleFluxoEmpresarial.Models.Compras
{
    public class CompraProduto : IBaseEntity
    {
        public string Modelo { get; set; }
        public string Serie { get; set; }
        public string Numero { get; set; }
        public int FornecedorId { get; set; }

        public int ProdutoId { get; set; }

        public Produto Produto { get; set; }

        public decimal Quantidade { get; set; }

        public decimal ValorUnitario { get; set; }

        public decimal? Desconto { get; set; }

        public decimal? IPI { get; set; }

        public decimal CustoUnitario { get; set; }
    }
}
